### PR TITLE
update tengine: add OneFlow, Horizon and Bitman model support; add op…

### PR DESCRIPTION
**Pull request by pierricklee**
add OneFlow, Horizon, and Bitman model support;
add tengine operators.
fix Reshape display.

### CODE REVIEW

File: source/tengine.js [hunk 1]
```diff
@@ -637,6 +642,7 @@ tengine.ModelFileReader = class {
         this._graphs = [];
         const subgraphOffsets = reader.uint32s();
         for (const subgraphOffset of subgraphOffsets) {
+            subgraphOffsets.length =1 ;
```
Comment (lutzroeder) [inline source/tengine.js:645]: Why is this required?

File: source/tengine.js [hunk 2]
```diff
@@ -548,8 +548,8 @@ tengine.ModelFileReader = class {
         register(20, 0, 'ReLU', [ 'f' ]);
         register(21, 0, 'ReLU6', []);
         register(22, 0, 'Reorg', [ 'i' ]);
-        register(23, 0, 'Reshape', [ 'i', 'i', 'i', 'i', 'i', 'i' ]);
-        register(23, 2, 'Reshape', [ 'i', 'i', 'i[]' ]);
+        // register(23, 0, 'Reshape', [ 'i', 'i', 'i', 'i', 'i', 'i' ]);
```
Comment (lutzroeder) [inline source/tengine.js:551]: Test regression. Run `node ./test/models.js tengine`.

File: source/tengine-metadata.json [hunk 3]
```diff
@@ -218,6 +218,21 @@
       { "name": "detect_num" }
     ]
   },
+  {
+    "name": "Reshape",
```
Comment (lutzroeder) [inline source/tengine-metadata.json:222]: Duplicate.

File: source/tengine-metadata.json [hunk 4]
```diff
@@ -866,6 +881,16 @@
       { "name": "type", "type": "int32", "default": 0 }
     ]
   },
+  {
+    "name": "Scatter",
```
Comment (lutzroeder) [inline source/tengine-metadata.json:885]: Sort alphabetically.

**Review by lutzroeder — CHANGES_REQUESTED** [review]

File: source/tengine.js (hunk 1 above)
Comment (pierricklee) [reply source/tengine.js:645]: Hi Lutz, 

This is for Tengine's subgraph strategy. Tengine sets the main subgraph as subgraph[0], other subgraphs[1-n] are cut from subgraph[0]. These subgraphs may have different data type, due to a hybrid-precision strategy. For example, subgraph[1] is float32 while subgraph[2] uses uint8, they are both sub-graph to subgraph[0]. 

So basically, the subgraph[0] is not actually a 'subgraph'. It is 'the graph'. 
Subgraphs shouldn't be shown when rendering, only 'the graph' will be shown.

[hybrid-precision subgraph  model.zip](https://github.com/lutzroeder/netron/files/6471719/hybrid-precision.subgraph.model.zip)

We can use this model to test. It's also an experimental feature in Tengine.
